equinix/errors.go

```diff
@@ -51,6 +53,17 @@ func convertToFriendlyError(errors Errors, resp *http.Response) error {
 	return er
 }
 
+func networkErrorOutput(err error, resp *http.Response) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	responseBody, readErr := io.ReadAll(resp.Body)
+	resp.Body.Close()
+	if readErr != nil {
+		diags = append(diags, diag.Diagnostic{Severity: 2, Summary: fmt.Sprintf("error parsing network request body: %s", readErr)})
+	}
+	diags = append(diags, diag.Diagnostic{Severity: 2, Summary: fmt.Sprintf("%s; \n %s", err, responseBody)})
```

Comment:
Can the responseBody be parsed into an Equinix API error object, and can the messages from that be included in the error message, rather than a raw json error? (or is that happening elsewhere)

Reply:
Are you talking about this type?: 
```
type ErrorResponse struct {
	StatusCode int
	Errors
	IsAPIError bool
}
```

The error response from Fabric is pretty different:
```
[{
    "errorCode":"EQ-3142601",
    "errorMessage":"Error while validating Authorization Key",
    "details":"Authorization key provided is not valid.",
    "correlationId":"pd6cVulZP2&JMxPsfDtIoiG#3",
    "additionalInfo":[
        {"property":"/zSide/accessPoint/authenticationKey"}
    ]
}]
```

For some reason I'm not quite seeing the connection at this point, but would be happy to attempt a similar pattern once it clicks. I don't think I've seen what the Terraform Error Output difference would be, but I agree that an improvement to it would be nice so that it isn't just a string of white text underneath the error.

Reply:
@displague I've updated the PR to align with your comments here. We don't have OpenAPIGenerator yet for Fabric-Go so we can't modify the SDK to include the formatMessaging there to do the same exact process as Metal errors. It's doing the message formatting after doing type assertion on the SDK's returned to get the details from Fabric-Go's ModelError type.

When @ctreatma is ready for other services to be added to https://github.com/equinix/equinix-sdk-go we will add Fabric to that with the OpenAPIGenerator and leave the old project on SwaggerCodegen. Then we will gradually migrate our resources to the grouped SDK during a refactor effort.

Thank you for the great feedback on getting this more aligned overall. We're still working towards convergence of development patterns but it's moving in that direction 👍